InvoiceLineItemCollection provided by getLines doesn't get same auth request options with autoPagingIterable

I'm using this library in a multithreaded environment against various keys granted via oauth by Stripe users for my app, so I provide the appropriate key in request options for each call I make. However, it seems like some of the models' methods don't copy these credentials on additional requests that are made.

e.g., I call Invoice.list passing my request options, and then on each invoice I call getLines and use the autoPagingIterable to access all invoice line items. However, for any invoice with enough line items such that an additional request is needed I get an exception like this:

```
java.lang.RuntimeException: Unable to lazy-load stripe objects
Caused by: com.stripe.exception.AuthenticationException: No API key provided. (HINT: set your API key using 'Stripe.apiKey = <API-KEY>'. You can generate API keys from the Stripe web interface. See https://stripe.com/api for details or email support@stripe.com if you have questions.
```

e: I'm using 1.41.0

--- a/src/main/java/com/stripe/model/StripeCollectionAPIResource.java
+++ b/src/main/java/com/stripe/model/StripeCollectionAPIResource.java
@@ -59,9 +59,24 @@ public abstract class StripeCollectionAPIResource<T extends HasId> extends APIRe
 	/**
 	 * Returns an iterable that can be used to iterate across all objects
 	 * across all pages. As page boundaries are encountered, the next page will
-	 * be fetch automatically for continued iteration.
+	 * be fetched automatically for continued iteration.
 	 */
 	public Iterable<T> autoPagingIterable() {
+		return autoPagingIterable(null, null);
+	}
+
+	public Iterable<T> autoPagingIterable(Map<String, Object> params) {
+		return autoPagingIterable(params, null);
+	}
+
+	public Iterable<T> autoPagingIterable(Map<String, Object> params, RequestOptions options) {
+		// This does have the side effect of manipulating the state of the
+		// given collection, which is an unfortunate side effect, but is quite
+		// likely a good practical trade-off in that these should only be used
+		// for auto pagination with that particular collection instance.
+		this.setRequestOptions(options);
+		this.setRequestParams(params);
+
 		return new PagingIterable<T>(this);
 	}
 

